src/main/java/org/alfresco/repo/content/metadata/AsynchronousExtractor.java

```diff
@@ -0,0 +1,537 @@
+                            enableStringTagging, carryAspectProperties, stringTaggingSeparators);
+
+                    if (logger.isTraceEnabled())
+                    {
+                        logger.trace("Extraction of Metadata from " + nodeRef + " complete " + changedProperties);
+                    }
+
+                    return null;
+                }, false, true));
+    }
+
+    private Map<String, Serializable> readMetadata(InputStream transformInputStream)
+    {
+        try
+        {
+            TypeReference<HashMap<String, Serializable>> typeRef = new TypeReference<HashMap<String, Serializable>>() {};
+            return jsonObjectMapper.readValue(transformInputStream, typeRef);
+        }
+        catch (IOException e)
+        {
+            logger.error("Failed to read metadata from transform result", e);
+            return null;
+        }
+    }
+
+    private String metadataToString(Map<String, Serializable> metadata)
+    {
+        Map<String, String> metadataAsStrings = AbstractMappingMetadataExtracter.convertMetadataToStrings(metadata);
+        try
+        {
+            return jsonObjectMapper.writeValueAsString(metadataAsStrings);
+        }
+        catch (JsonProcessingException e)
+        {
+            logger.error("Failed to save metadata as Json", e);
+            return null;
+        }
+    }
+
+    private OverwritePolicy removeOverwritePolicy(Map<String, Serializable> map, String key, OverwritePolicy defaultValue)
+    {
+        Serializable value = map.remove(key);
+        if (value == null)
+        {
+            return defaultValue;
+        }
+        try
+        {
+            return OverwritePolicy.valueOf((String)value);
+        }
+        catch (IllegalArgumentException|ClassCastException e)
+        {
+            logger.error(key + "=" + value + " is invalid");
+            return null;
+        }
+    }
+
+    private Boolean removeBoolean(Map<String, Serializable> map, Serializable key, boolean defaultValue)
+    {
+        @SuppressWarnings("SuspiciousMethodCalls") Serializable value = map.remove(key);
+        if (value != null &&
+            (!(value instanceof String) ||
+             (!(Boolean.FALSE.toString().equals(value) || Boolean.TRUE.toString().equals(value)))))
+        {
+            logger.error(key + "=" + value + " is invalid. Must be " + Boolean.TRUE + " or " + Boolean.FALSE);
+            return null; // no flexibility of parseBoolean(...). It is just invalid
+        }
+        return value == null ? defaultValue : Boolean.parseBoolean((String)value);
+    }
+
+    private List<String> removeTaggingSeparators(Map<String, Serializable> map, String key, List<String> defaultValue)
+    {
+        Serializable value = map.remove(key);
+        if (value == null)
+        {
+            return defaultValue;
+        }
+        if (!(value instanceof String))
+        {
+            logger.error(key + "=" + value + " is invalid.");
+            return null;
```

Comment:
It could but I wanted to use null to indicate an error rather than an empty list. Could throw an exception and catch it one level up, but that appeared to be over kill. This pattern is used all over the repo code.